exporter/prometheusremotewriteexporter/wal.go

```diff
@@ -66,13 +68,17 @@ func newWAL(walConfig *WALConfig, exportSink func(context.Context, []*prompb.Wri
 		return nil, errNilConfig
 	}
 
-	return &prweWAL{
+	wal := prweWAL{
 		exportSink: exportSink,
 		walConfig:  walConfig,
 		stopChan:   make(chan struct{}),
+		rNotify:    make(chan struct{}),
 		rWALIndex:  &atomic.Uint64{},
 		wWALIndex:  &atomic.Uint64{},
-	}, nil
+		log:        zap.NewNop(),
```

Comment:
There is a logger available in the `TelemetrySettings`. But adding a `zap.NewNop()` logger here, should be equal to the state before without any logger, right?

Not that I have a strong opinion about it - I just would like to resolve it at some point :smile:

Reply:
I think it is more the point of, if we are adding a logger, we need some way of providing a configuration for it.

Reply:
the logger is taken from the context once `run()` is called, but we need a place to store it for long running operations.

not setting this var leads to a nil pointer that panics, so its `NewNop()` instead to avoid that. afaict this is never called before run() anways